nginx.tmpl

```diff
@@ -750,6 +753,7 @@ server {
         try_files $uri =404;
         break;
     }
+        {{- end }}
```

Comment:
```suggestion
```

Reply:
There are broken use cases with both defaults:

**true**
acme_companion configures an extraneous HTTP Challenge location block which is not valid.

**false**
Use cases where certificates are not generated with acme_companion loose HTTP Challenge configuration and may fail to renew their certificates.

Why not fixing acme_companion instead?

EDIT: The easiest way would be to keep the default as `true` and update acme_companion so it wouldn't generate the HTTP Challenge configuration by default (but for standalone certificates).

EDIT1: moving this comment with was on the wrong thread.

Reply:
How about having 3 possible values for `ACME_HTTP_CHALLENGE`?
* `legacy`: previous incomplete behavior
* `true`: generate ACME HTTP Challenge location blocks when `HTTPS_METHOD` is in [ `redirect`, `noredirect` ] (current default behavior) 
* `false`: do not generate ACME HTTP Challenge location blocks

And we set the new default to `legacy`.

Reply:
The `legacy` / `true` /  `false` solution with `legacy` being the default (until we change acme-companion default behaviour, which I plan on doing soon) seems perfect to me, if it's not too much additional work for you.

> Why not fixing acme_companion instead?

Because I'm a bit short in time this week 😬 but that's planned, yes.

(by the way regarding acme-companion and it's location tinkering, I tried to automatically skip location configuration when not required a few years ago, I could not come up with anything else than very hacky and fragile solutions that did not even work in the end. Manually enabling or disabling location configuration as you suggest seem to be the only realistic way).

Reply:
Minor request: could we use `ACME_HTTP_CHALLENGE_LOCATION` instead of `ACME_HTTP_CHALLENGE` ?

Reply:
Done.